src/record/index.js

```diff
@@ -7,25 +7,31 @@ const errcode = require('err-code')
  */
 class Record {
   /**
-   * @constructor
-   * @param {String} domain signature domain
-   * @param {Uint8Array} codec identifier of the type of record
+   * @class
+   * @param {string} domain - signature domain
+   * @param {Uint8Array} codec - identifier of the type of record
    */
   constructor (domain, codec) {
     this.domain = domain
     this.codec = codec
   }
 
+  // eslint-disable-next-line
   /**
    * Marshal a record to be used in an envelope.
+   *
+   * @returns {Uint8Array}
    */
   marshal () {
     throw errcode(new Error('marshal must be implemented by the subclass'), 'ERR_NOT_IMPLEMENTED')
   }
 
+  // eslint-disable-next-line
```

Comment:
The issue here is the return below. We use this in libp2p generically in the https://github.com/libp2p/js-libp2p/blob/master/src/record/envelope/index.js Envelope.seal(). If we do not set here the types returned it would be void and cause issues there